### PR TITLE
Ben/powerup

### PR DESCRIPTION
**Pull request by UnwiseCheese98**
Pull request checklist:
- [x] Merge conflicts have been resolved
- [x] The UML diagrams have been updated accordingly
- [x] My changes respect the Google Java Style

### CODE REVIEW

**Comment by UnwiseCheese98** [comment]
[changestouml2.txt](https://github.com/Frinksy/PixelBoat/files/5937727/changestouml2.txt)

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java [hunk 1]
```diff
@@ -146,6 +150,32 @@ public void hasCollidedEnergy() {
     }
   }
 
+  /**
+   * Applies collision with rotation bonus.
+   *
```
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java:155]: there's an extra space on that line that causes checkstyle to fail

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java [hunk 2]
```diff
@@ -146,6 +150,32 @@ public void hasCollidedEnergy() {
...
+    if (rotationSpeed < 2.75) {
+      rotationSpeed += 0.2;
+    } else {
+      rotationSpeed = 3;
+    }
+  }
+
+  /**
+   * Applies collision with rotation bonus.
+   *
```
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java:168]: there's an extra space on that line that causes checkstyle to fail

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java (hunk 1 above)
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java:155]: there's an extra space on that line that causes checkstyle to fail

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java [hunk 3]
```diff
@@ -137,6 +139,8 @@ public void hasCollidedHealth() {
 
   /**
    * Applies collision with energy bonus.
+   *
```
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java:142]: there's an extra space on that line that causes checkstyle to fail

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java [hunk 4]
```diff
@@ -126,6 +126,8 @@ public void hasCollidedSpeed() {
 
   /**
    * Applies collision with health bonus.
+   *
```
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java:129]: there's an extra space on that line that causes checkstyle to fail

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpDrag.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
+package com.teamonehundred.pixelboat.entities;
+
+import com.badlogic.gdx.math.Rectangle;
+import com.badlogic.gdx.math.Vector2;
+
+/**
+ * Represents the drag reduction power up
```
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpDrag.java:7]: needs full stop

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpEnergy.java [hunk 6]
```diff
@@ -4,7 +4,7 @@
 import com.badlogic.gdx.math.Vector2;
 
 /**
- * PowerUpEnergy.
+ * Represents the Energy power up
```
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpEnergy.java:7]: needs a full stop

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpHealth.java [hunk 7]
```diff
@@ -4,7 +4,7 @@
 import com.badlogic.gdx.math.Vector2;
 
 /**
- * Represents the simple branch obstacle.
+ * Represents the Health power up
```
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpHealth.java:7]: needs a full stop

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpHealth.java [hunk 8]
```diff
@@ -23,15 +23,14 @@ public PowerUpHealth(float x, float y) {
...
-   * <p>Creates a new collision bounds object representing the current position of this branch.
+   * <p>Creates a new collision bounds object representing the current position of this power up.
    * See the collision bounds visualisation folder in assets for a visual representation.
    *
-   * @return CollisionBounds of branch obstacle
-   * @author James Frost
-   * @author William Walton
+   * @return CollisionBounds of power up object
+   * @author Ben Dunbar
    */
```
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpHealth.java:34]: Block comment has incorrect indentation level 3, expected is 2,

File: Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpRotation.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
+package com.teamonehundred.pixelboat.entities;
+
+import com.badlogic.gdx.math.Rectangle;
+import com.badlogic.gdx.math.Vector2;
+
+/**
+ * Represents the rotation power up
```
Comment (Frinksy) [inline Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpRotation.java:7]: needs a full stop

**Review by Frinksy — CHANGES_REQUESTED** [review]
You want to address the errors below:
```
2021-02-06T21:37:13.8902768Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java:130: Javadoc tag '@author' should be preceded with an empty line. [RequireEmptyLineBeforeBlockTagGroup]
2021-02-06T21:37:13.8903935Z 
2021-02-06T21:37:13.8906794Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java:143: Javadoc tag '@author' should be preceded with an empty line. [RequireEmptyLineBeforeBlockTagGroup]
2021-02-06T21:37:13.8908070Z > Task :core:checkstyleMain
2021-02-06T21:37:13.8909968Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java:156: Javadoc tag '@author' should be preceded with an empty line. [RequireEmptyLineBeforeBlockTagGroup]
2021-02-06T21:37:13.8912465Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/Boat.java:169: Javadoc tag '@author' should be preceded with an empty line. [RequireEmptyLineBeforeBlockTagGroup]
2021-02-06T21:37:13.8914476Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpDrag.java:6: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
2021-02-06T21:37:13.8916278Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpEnergy.java:6: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
2021-02-06T21:37:13.8918173Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpHealth.java:6: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
2021-02-06T21:37:13.8929884Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpHealth.java:26:4: Block comment has incorrect indentation level 3, expected is 2, indentation should be the same level as line 35. [CommentsIndentation]
2021-02-06T21:37:13.8932328Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpRotation.java:6: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
2021-02-06T21:37:13.8934388Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpRotation.java:26:4: Block comment has incorrect indentation level 3, expected is 2, indentation should be the same level as line 35. [CommentsIndentation]
2021-02-06T21:37:13.8936442Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpSpeed.java:6: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
2021-02-06T21:37:13.8938445Z [ant:checkstyle] [ERROR] /home/runner/work/PixelBoat/PixelBoat/Implementation/core/src/main/java/com/teamonehundred/pixelboat/entities/PowerUpSpeed.java:26:4: Block comment has incorrect indentation level 3, expected is 2, indentation should be the same level as line 35. [CommentsIndentation]
```

Test with `./gradlew checkstyleMain`

**Pull request closed without merging**